glfs.c

```diff
@@ -566,7 +567,13 @@ static void tcmu_glfs_close(struct tcmu_device *dev)
 	free(gfsp);
 }
 
+#if GFAPI_VERSION < 741
 static void glfs_async_cbk(glfs_fd_t *fd, ssize_t ret, void *data)
+#elif GFAPI_VERSION < 760
+static void glfs_async_cbk(glfs_fd_t *fd, ssize_t ret, struct stat *prestat, struct stat *poststat, void *data)
```

Comment:
I do not think this variation exists? GlusterFS 4.1 and 5 have

```
typedef void (*glfs_io_cbk)(glfs_fd_t *fd, ssize_t ret, void *data);
````

Reply:
yeah, also the same, just checked this by using the 'git tag --contains SHA', which has been reverted in all the release branches.

Will update it.

Reply:
```
commit 384562b294e9a7847403961e878a4daa0fff33eb
Author: Kinglong Mee <kinglongmee@gmail.com>
Date:   Thu Feb 1 11:20:37 2018 +0800

    gfapi: return pre/post attributes at callback for glfs api
    
    Updates: #389
    Change-Id: Ic71632722effe4b8855d5de3e65688efd9afe1e3
    Signed-off-by: Kinglong Mee <mijinlong@open-fs.com>

diff --git a/api/src/glfs.h b/api/src/glfs.h
index e2f2a2c..c8b261e 100644
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -499,7 +499,9 @@ int glfs_set_xlator_option (glfs_t *fs, const char *xlator, const char *key,
   in a common callback function.
 */
 
-typedef void (*glfs_io_cbk) (glfs_fd_t *fd, ssize_t ret, void *data);
+typedef void (*glfs_io_cbk34) (glfs_fd_t *fd, ssize_t ret, void *data);
+typedef void (*glfs_io_cbk) (glfs_fd_t *fd, ssize_t ret, struct stat *prestat,
+                             struct stat *poststat, void *data);
 
 // glfs_{read,write}[_async]
 

```